File: Ipclass.c
#include <stdio.h>
#include <string.h>
#include <stdlib.h>

void findClassAndType(char* ipAddress) {
    int octets[4];
    char* token = strtok(ipAddress, ".");
    
    // Parse the IP address into octets
    for (int i = 0; i < 4; i++) {
        if (token == NULL) {
            printf("Invalid IP address format.\n");
            return;
        }
        octets[i] = atoi(token);
        token = strtok(NULL, ".");
    }
    
    int firstOctet = octets[0];
    char ipClass;
    
    // Determine the IP class
    if (firstOctet >= 1 && firstOctet <= 127) {
        ipClass = 'A';
    } else if (firstOctet >= 128 && firstOctet <= 191) {
        ipClass = 'B';
    } else if (firstOctet >= 192 && firstOctet <= 223) {
        ipClass = 'C';
    } else if (firstOctet >= 224 && firstOctet <= 239) {
        ipClass = 'D';
    } else {
        ipClass = 'E';
    }

    // Check if the IP is private
    int isPrivate = 0;
    if ((ipClass == 'A' && firstOctet == 10) ||
        (ipClass == 'B' && firstOctet == 172 && octets[1] >= 16 && octets[1] <= 31) ||
        (ipClass == 'C' && firstOctet == 192 && octets[1] == 168)) {
        isPrivate = 1;
    }

    // Print the results
    printf("IP Address: %s\n", ipAddress);
    printf("Class: %c\n", ipClass);
    printf("Type: %s\n", isPrivate ? "Private" : "Public");
}

int main() {
    char ipAddress[16];
    printf("Enter an IP address: ");
    scanf("%15s", ipAddress);

    findClassAndType(ipAddress);

    return 0;
}

File: IpClass.java
import java.util.Scanner;

public class IpClass {

    public static void findClassAndType(String ipAddress) {
        String[] octets = ipAddress.split("\\.");
        int firstOctet = Integer.parseInt(octets[0]);

        // Determine the IP class
        char ipClass;
        if (firstOctet >= 1 && firstOctet <= 127) { // Corrected range to start from 1 instead of 0
            ipClass = 'A';
        } else if (firstOctet >= 128 && firstOctet <= 191) {
            ipClass = 'B';
        } else if (firstOctet >= 192 && firstOctet <= 223) {
            ipClass = 'C';
        } else if (firstOctet >= 224 && firstOctet <= 239) {
            ipClass = 'D';
        } else {
            ipClass = 'E';
        }

        // Check if the IP is private
        boolean isPrivate = false;
        if ((ipClass == 'A' && firstOctet == 10) ||
            (ipClass == 'B' && firstOctet == 172 && Integer.parseInt(octets[1]) >= 16 && Integer.parseInt(octets[1]) <= 31) ||
            (ipClass == 'C' && firstOctet == 192 && Integer.parseInt(octets[1]) == 168)) {
            isPrivate = true;
        }

        // Print the results
        System.out.println("IP Address: " + ipAddress);
        System.out.println("Class: " + ipClass);
        System.out.println("Type: " + (isPrivate ? "Private" : "Public"));
    }

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        System.out.print("Enter an IP address: ");
        String ipAddress = scanner.nextLine();

        findClassAndType(ipAddress);
        scanner.close();
    }
}
